modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -259,18 +259,15 @@ private void closeAll(boolean isArchiving) {
         isOpened = false;
     }
 
-    private void closeDataOutputStreams() {
+    private void closeAllHdfsFiles() {
         for (Map.Entry<String, HdfsFile> entry : openedFiles.entrySet()) {
-            FSDataOutputStream outputStream = entry.getValue().getFsDataOutputStream();
-            if (outputStream != null) {
-                try {
-                    logger.debug(String.format("Closing file: %s", entry.getKey()));
-                    outputStream.close();
-                } catch (IOException e) {
-                    printStackTrace(e);
-                }
+            HdfsFile hdfsfile = entry.getValue();
+            try {
+                logger.debug(String.format("Closing file: %s", entry.getKey()));
+                hdfsfile.close();
```

Comment:
thanks.
At destination side's send() method we don't remove the file neither on close nor after archive.
However since after close() we need to call open(), `openedFiles` will be re-created, so this is fine.